java/ql/src/Security/CWE/CWE-200/AndroidWebViewSettingsFileAccess.qhelp

```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE qhelp PUBLIC
+ "-//Semmle//qhelp//EN"
+ "qhelp.dtd">
+<qhelp>
+  <overview>
+    <p>
+      File access in an Android WebView can expose the device's file system to
+      the JavaScript running in the WebView. If there are vulnerabilities in the
+      JavaScript, file access may allow an attacker to access or steal the
+      user's data.
+    </p>
+  </overview>
+
+  <recommendation>
+    <p>When possible, you should disallow file access by setting the following settings to <code>false</code>:</p>
+
+    <ul>
+      <li><code>setAllowFileAccess</code></li>
+      <li><code>setAllowFileAccessFromFileURLs</code></li>
+      <li><code>setAllowUniversalAccessFromFileURLs</code></li>
+    </ul>
+  </recommendation>
+
+  <example>
+    <p>In the following (bad) example, the WebView is configured with the settings
+    which would allow local file access.</p>
+
+    <sample src="WebViewFileAccessUnsafe.java"/>
+
+    <p>In the following (good) example, the WebView is configured to disallow file access.</p>
```

Comment:
I think an example of how to use [`AssetLoader`](https://developer.android.com/reference/androidx/webkit/WebViewAssetLoader) as a safe alternative would be interesting as well.